tests/unit/test_main.py

```diff
@@ -1016,3 +1016,582 @@ def test_match_and_apply(self, flow, args, resp):
+                        "dl_vlan": 201
+                        },
+                    "instructions": [{
+                        "instruction_type": "apply_actions",
+                        "actions": [{
+                            "action_type": "send_report"
+                        }]},
+                        {
+                            "instruction_type": "goto_table",
+                            "table_id": 2
+                        }
+                    ],
+                    "table_id": 0,
+                    "table_group": "evpl",
+                    "priority": 20000,
+                    }, },
+                {"flow": {
+                    "match": {
+                        "in_port": 20,
+                        "dl_vlan": 201
+                        },
+                    "instructions": [{
+                        "instruction_type": "apply_actions",
+                        "actions": [{
+                            "action_type": "pop_int"
+                        }, {
+                            "action_type": "set_vlan",
+                            "vlan_id": 200
+                        }, {
+                            "action_type": "output",
+                            "port": 16
+                        }]}
+                    ],
+                    "table_id": 2,
+                    "table_group": "base",
+                    "priority": 20000,
+                    }, },
+                ]}
+        payload = [
+            {"trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 15
+                    },
+                "eth": {"dl_vlan": 201}
+                }},
+            {"trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 16
+                    },
+                "eth": {"dl_vlan": 200}
+                }},
+            {"trace": {
+                "switch": {
+                    "dpid": "00:00:00:00:00:00:00:01",
+                    "in_port": 20
+                    },
+                "eth": {"dl_vlan": 201}
+                }}
+        ]
+
+        resp = await self.api_client.put(self.traces_endpoint, json=payload)
+        assert resp.status_code == 200
+        current_data = resp.json()
+        result = current_data["result"]
+
+        result_ = result[0]
+        assert result_[0]['dpid'] == '00:00:00:00:00:00:00:01'
+        assert result_[0]['port'] == 15
+        assert result_[0]['type'] == 'last'
+        assert result_[0]['vlan'] == 201
+        assert result_[0]['out']['port'] == 19
+        assert result_[0]['out']['vlan'] == 201
+
+        result_ = result[1]
+        assert result_[0]['dpid'] == '00:00:00:00:00:00:00:01'
+        assert result_[0]['port'] == 16
+        assert result_[0]['type'] == 'last'
+        assert result_[0]['vlan'] == 200
+        assert result_[0]['out']['port'] == 17
```

Comment:
![20231006_144838](https://github.com/kytos-ng/sdntrace_cp/assets/1010796/e662c7f3-da67-400b-b783-43dc67d90081)

Same as my prior comment, except it should terminate on port 15, check out the attached diagram

Reply:
Ok.